amesbah_edge_merging_proposal.md

```diff
@@ -0,0 +1,417 @@
+# Knowledge Graph Edge Merging
+
+
+Based on discussions, it has been decided that some level of merging is necessary with the Translator. This is a proposal for the conditions underwhich knowledge graph edges can be merged. Subsequent discussions and proposals will be required to determine the necessary TRAPI schema changes so that no information is lost during merging.
+
+## Proposal
+
+Edges in the knowledge graph may be merged if they share the following properties:
+* “subject”: The two edges must share the same subject. To facilitate this, node normalization may be required, in case two KP’s utilize different CURIE’s to refer to the same entity.
+* “object”: The two edges must share the same object. To facilitate this, node normalization may be required, in case two KP’s utilize different CURIE’s to refer to the same entity.
+* “predicate”: The two edges must use the same biolink predicate. Descendent predicates should not be merged with ancestral predicates.
+* “qualifiers”: Any qualifiers in the candidate edges must be identical.
+
+Additionally, certain attributes must also be the same:
+* "biolink: original_knowledge_source"
+OR
+* "biolink: primary_knowledge_source"
+
+All edges must have one (and only one) of these attributes listed, and to merge two edges, they must share the same value for the attribute.
```

Comment:
re: "must have one (and only one) of these attributes listed" . . . there will be 0, 1, or many qualifiers on a  given Edge . . . so the one and only one rule cannot apply here.